### PR TITLE
NN-3900: Use reported adjudications table instead of NOMIS when getting data

### PR DESCRIPTION
**Pull request by sp-watson**
There is about 60 LoC changed in the main source - the rest (mostly deletions) are test changes, esp integration test changes

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt [hunk 1]
```diff
@@ -165,32 +166,33 @@ class DraftAdjudicationIntTest : IntegrationTestBase() {
...
-      .jsonPath("$.prisonerNumber").isEqualTo("A12345")
-      .jsonPath("$.bookingId").isEqualTo("1")
-      .jsonPath("$.dateTimeReportExpires").isEqualTo("2010-11-15T10:00:00")
-      .jsonPath("$.incidentStatement.statement").isEqualTo("new statement")
-      .jsonPath("$.incidentDetails.dateTimeOfIncident").isEqualTo("2010-11-12T10:00:00")
-      .jsonPath("$.incidentDetails.handoverDeadline").isEqualTo("2010-11-15T10:00:00")
-      .jsonPath("$.incidentDetails.locationId").isEqualTo(721850)
+      .jsonPath("$.adjudicationNumber").isEqualTo(IntTestData.DEFAULT_ADJUDICATION.adjudicationNumber)
+      .jsonPath("$.prisonerNumber").isEqualTo(IntTestData.DEFAULT_ADJUDICATION.prisonerNumber)
+      .jsonPath("$.bookingId").isEqualTo(1) // From PrisonAPIMockServer.stubPostAdjudication
```
Comment (sp-watson) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt:183]: We really ought to address this - the bookingId should be part of the test data

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt [hunk 2]
```diff
@@ -165,32 +166,33 @@ class DraftAdjudicationIntTest : IntegrationTestBase() {
   @Test
   fun `complete draft adjudication`() {
     prisonApiMockServer.stubPostAdjudication()
-    bankHolidayApiMockServer.stubGetBankHolidays()
+    val intTestData = IntTestData(webTestClient, jwtAuthHelper, bankHolidayApiMockServer, prisonApiMockServer)
 
-    val draftAdjudicationResponse = dataAPiHelpers().startNewAdjudication(dateTimeOfIncident = DATE_TIME_OF_INCIDENT)
-    dataAPiHelpers().addIncidentStatement(draftAdjudicationResponse.draftAdjudication.id, "test statement")
```
Comment (sp-watson) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt:171]: I am slowly removing all refs to dataAPiHelpers() in place of IntTestData, as the latter is more coherent

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt [hunk 3]
```diff
@@ -165,32 +166,33 @@ class DraftAdjudicationIntTest : IntegrationTestBase() {
...
+    val firstDraftUserHeaders = setHeaders(username = IntTestData.DEFAULT_ADJUDICATION.createdByUserId)
```
Comment (sp-watson) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt:171]: We ought to decide on a better way of describing the setup rather than this idiomatic use of IntTestData

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/IntTestData.kt [hunk 4]
```diff
@@ -23,13 +23,38 @@ class IntTestData(
     const val DEFAULT_AGENCY_ID = "MDI"
     const val DEFAULT_CREATED_USER_ID = "B_MILLS"
     val DEFAULT_DATE_TIME_OF_INCIDENT = LocalDateTime.of(2010, 11, 12, 10, 0)
+    const val DEFAULT_DATE_TIME_OF_INCIDENT_TEXT = "2010-11-12T10:00:00"
+    const val DEFAULT_HANDOVER_DEADLINE_ISO_STRING = "2010-11-15T10:00:00"
```
Comment (sp-watson) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/IntTestData.kt:27]: I am working on deriving the ISO String from the date - but that'll be a different commit

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/SpringOauthIntegrationTest.kt [hunk 5]
```diff
@@ -4,30 +4,41 @@ import com.github.tomakehurst.wiremock.client.WireMock
...
+  private lateinit var intTestData: IntTestData
+  private lateinit var draftCreationResponseForAdjudication1: DraftAdjudicationResponse
+  private lateinit var draftCreationResponseForAdjudication2: DraftAdjudicationResponse
+
   @BeforeEach
   fun beforeEach() {
-    prisonApiMockServer.stubGetAdjudication()
+    intTestData = IntTestData(webTestClient, jwtAuthHelper, bankHolidayApiMockServer, prisonApiMockServer)
+
+    draftCreationResponseForAdjudication1 = intTestData.startNewAdjudication(IntTestData.ADJUDICATION_1)
```
Comment (sp-watson) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/SpringOauthIntegrationTest.kt:20]: We need to make 2 adjudications ready for reporting so we can check the prison API calls use the correct creds

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt (hunk 2 above)
Comment (richardpopple) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt:171]: Cool

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/IntTestData.kt (hunk 4 above)
Comment (richardpopple) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/IntTestData.kt:27]: Cool

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt (hunk 1 above)
Comment (richardpopple) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt:183]: I'm being a bit slow I'm not sure I understand this comment

Comment (sp-watson) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt:183]: The code should read something like `isEqualTo(IntTestData.DEFAULT_ADJUDICATION.bookingId)` instead of being hard-coded - and that means adding the `bookingId` to the test data sets

**Review by richardpopple — APPROVED** [review]
